Upgrade Child Theme to version 2

matt-bernhardt opened the pull request:
(Edit) This has turned into a massive pull request that updates nearly the entirety of the Child theme, and when merged will become version 2 of that theme.

matt-bernhardt committed:
Adds Travis and CodeClimate configuration for code analysis

PBruk committed:
New - supplementary - version committed.

PBruk committed:
Restructure Child header to eliminate box1 & box2; add ability to load full-width bg images; add to and consolidate CSS to reflect styling changes to match designs.

PBruk committed:
Change styling of child navbar to reflect new designs.

PBruk committed:
Change sidebar styling to reflect new designs.

PBruk committed:
Page template allows for full-width content, without sidebar.

PBruk committed:
Add page-template and template-part to allow for widgetized sections below the content.

PBruk committed:
Add styling for staff widgets per design work.

PBruk committed:
Add general padding and margin styling for standard page layouts.

PBruk committed:
Add styling for sidebar staff widget and other related adjustments per design comps.

PBruk committed:
Standardizes sidebar font-size to include .textwidget class per design comps.

PBruk committed:
Add/adjust styling for Full-Width template as per design comps.

PBruk committed:
Adjust font-size to match design comps.

PBruk committed:
Adds classes to give a set of buttons margins and make them inline.

PBruk committed:
Adjusted styling for galleries (desktop) to match design mockup; Reordered and added comments for various CSS rulesets.

PBruk committed:
Add styling for galleries on mobile > 480px devices.

PBruk committed:
Adjust spacing between a group of buttons; Fix layout on mobile, so buttons don't break.

PBruk committed:
Remove rule that hides top border for sidebar-widget; reduce top padding.

PBruk committed:
Remove max-width from buttons on mobile, to ensure equal length.

PBruk committed:
Adjust which selectors control content padding; adjust padding on mobile devices.

PBruk committed:
Extends gallery images and captions full-width on mobile > 480px.

PBruk committed:
Add bottom-margin removed in a previous commit.

PBruk committed:
Adjust padding to line up with other elements on page - per edits.

PBruk committed:
Adjust navbar to line up with other elements on page - per edits; fix resulting scroll on mobile devices.

PBruk committed:
Adjust header gradients, titles; also make adjustments to various media screen sizes.

PBruk committed:
Adjust padding and bottom margin - per edits.

PBruk committed:
Adjust page-header per mockups/edits.

PBruk committed:
Adjust content h3 & p layout - per design comps/edits.

PBruk committed:
Adjust padding of content h2 - per mockups/edits.

PBruk committed:
Remove unnecessary top-margin.

PBruk committed:
Add classes to style specific widget-types.

PBruk committed:
Adjust spacing - per edits.

PBruk committed:
Reverse direction of gradient - per edits.

PBruk commented:
@matt-bernhardt can you please do a review of my PR for this branch? Apologies, in advance, for the volume of commits...

matt-bernhardt commented on page-full-width.php:
```diff
@@ -0,0 +1,33 @@
+<?php
+/**
+ Template Name: Full Width
```
This line needs an asterisk before the text starts

matt-bernhardt commented:
I'm reviewing the changes now, and will add comments where appropriate. This is the version running on Dev, correct?

matt-bernhardt commented on style.css:
```diff
@@ -1,265 +1,275 @@
+@import "../libraries/style.css";
```
The Wordpress Codex talks about parent themes being loaded via functions.php - see MITLibraries/courtyard-blog#23 for more info

PBruk commented:
This version is running on Dev, yes.

matt-bernhardt commented:
Are you still piecing together changes with more to come, or is this all of them? I ask because I don't see any changes to functions.php yet - and I thought there would be to define widget regions.

PBruk commented:
This is an ongoing process. As far as widgets go, I am inheriting the widget function from Parent and I am calling the function in the - new - content-widgetized.php file, which itself is being called by the - new - page-widgetized.php file.

Probably, tho, once we decide what we want those areas to be titled, on the backend, since they are currently just first front page widget area and second front page widget area, we should add the function to the functions.php in Child and rename it there.

matt-bernhardt commented:
Yeah, the way things are called now is producing duplicate ID values (both regions have `id="secondary"`) which isn't valid HTML - and may screw up styling attempts down the road.

I'd recommend sorting out what the regions are going to be called, and building out the relevant functions, sooner rather than later.

PBruk commented:
Okay, gotcha... I'll jump on that next'ish. Thanks for the head's up. Didn't see that!

PBruk committed:
Remove unused classes, standardize template structure, change camel-case to hyphens; adjust stylesheet to reflect changes; remove top-border on buttons in sidebar.

PBruk committed:
Adjust rules for content and sidebar width on mobile devices.

PBruk committed:
Unregister unused Parent sidebars; register new widgetized sidebar area; adjust template to call for new sidebar; create new sidebar template.

PBruk committed:
Adjust dimensions of customHeader; adjust dimensions of thumbnails.

PBruk committed:
When unregistering sidebars in Child theme, ALL sidebars should be unregistered, and then re-registered. In previous commit, the Main Sidebar remained registered.

PBruk committed:
Adjust formatting; remove flex-slider.

PBruk committed:
Adjust call to new main sidebar.

PBruk committed:
Retrieve image from post (to show beside excerpts).

PBruk committed:
Border rule should be handled by separate class.

PBruk committed:
Create new Front-Page templates to display sticky post; add styling for sticky post and other flex-box content.

PBruk committed:
Create class to overwrite max-width for buttons, useful for sidebar buttons, which need full-width.

PBruk committed:
Function allows for custom excerpt lengths.

PBruk committed:
Calls new excerpt function.

PBruk committed:
Add directory to add custom templates to for use with the ultimate posts widget; Add first custom template.

PBruk committed:
Add provisional styling for custom UPW template.

PBruk committed:
Move $custom_fields option to match design layout.

PBruk committed:
Remove extra padding around entry-content on mobile screens.

PBruk committed:
Adjust styling of elements that use flexbox for mobile screens.

PBruk committed:
Adjust (secondary) button layout for mobile screens.

PBruk committed:
Increase font-size of buttons.

PBruk committed:
Style sidebar widget social-media icons to match color in mockup.

matt-bernhardt committed:
Adds custom field treatment for event dates in UFW template

PBruk committed:
Make div a span so that fields will display inline

PBruk committed:
Per @frances' edits: many things, much of which was completely removing !important from the file; still have plenty of adjustments to fix what was lost.

PBruk committed:
Add class to target element without using !important.

PBruk committed:
Add class to target element without using !important.

PBruk committed:
Add class to target element without using !important.

PBruk committed:
Add !important back because gallery CSS/html is constructed in such a way that it is necessary; future project: to fix.

PBruk committed:
More styling to replace !important in gallery.

PBruk committed:
Open dropdown submenu on child nav. Needs further testing (on Android).

PBruk committed:
Style custom event date fields to match mockups and events on /news.

PBruk committed:
Make parent menu show active link if ancestor is currently linked.

PBruk committed:
Add flex styling to staff-widgets on mobile screens.

PBruk committed:
Remove already declared styling.

PBruk committed:
Provisional exhibits CPT styling.

PBruk committed:
Add padding to full-width pages on mobile screens.

PBruk committed:
Change 'aside' to 'div' on bottom widget area; adjust exhibit thumbnail size; change default image for $first_img.

PBruk committed:
Add/adjust styling for /exhibits site.

PBruk committed:
Remove

PBruk committed:
Undo last commit.

PBruk committed:
Adjust display of spans according to parent widget div.

PBruk committed:
Adjust padding of container.

PBruk committed:
Adjust spacing and placement of category initial and name for /exhibits on mobile screens.

PBruk committed:
Custom template to display Exhibit post excerpts when using Ultimate Posts Widget.

PBruk committed:
Page and template-part to display exhibit post excerpts per mockups.

PBruk committed:
Add extra top margin to sidebar when using exhibits page template.

PBruk committed:
Formatting changes. (Template might not be necessary, however.)

PBruk committed:
Reformat template; correct miscopied php.

PBruk committed:
Remove !important.

PBruk committed:
Refactor, reformat styling for /exhibits posts feed.

PBruk committed:
Remove styling.

PBruk committed:
Refactor, reformat styling /exhibits post feed on mobile.

PBruk committed:
Set header images to load as background images via CSS.

PBruk committed:
Add styling for new header image layout.

PBruk committed:
Adjust line-height of excerpt-title.

PBruk committed:
Home page template for /exhibits site.

PBruk committed:
Rename page template for display feed on /exhibits site.

PBruk committed:
Add new home page template for /exhibits site.

PBruk committed:
Deprecated template.

PBruk committed:
Deprecated template.

PBruk committed:
Remove slider template and its CSS.

PBruk committed:
Add extra margin to top of sidebar when using page-exhibits-display template.

PBruk committed:
Add/adjust styling for /exhibits display feed.

PBruk committed:
Adjust line-height and bottom padding of single.php header h2.

PBruk committed:
Template-part deprecated, no longer in use.

PBruk committed:
Template-part renamed, removed.

PBruk committed:
Original /exhibits content template-part split into 3 for display purposes; might revisit at a further date to consolidate.

PBruk committed:
Change template header to reflect proper information regarding name and use.

PBruk committed:
Adjust template header and remove code leftover from archive template structure.

PBruk committed:
Add template for full posts on /exhibits site.

PBruk committed:
Remove experimental template for /exhibits site.

PBruk committed:
Add category templates for /exhibits site.

PBruk committed:
Add styling for header content in single posts.

PBruk committed:
Remove camel case of selector class.

PBruk committed:
Add styling for custom fields when displayed on single-exhibits.php.

PBruk committed:
Add bottom margin to .excerpt-post in case there are multiple posts; add top margin to sidebars on category pages.

PBruk committed:
Add sticky posts.

PBruk committed:
Allow more sticky posts on page.

PBruk committed:
Change name of template used as selector.

PBruk committed:
Rename template headers to proper Category name.

PBruk committed:
Allow category templates to load header_image to match home page automatically since header can't be chosen manually.

PBruk committed:
Adjust height and width of header image.

PBruk committed:
Fixes #8, pursuant to WP best practices that suggests stylesheets be loaded via functions file, not imported.

PBruk committed:
Adds border to hr and adjust size and weight of Child H4.

PBruk committed:
Add clear to HR for better divide sections with image floats.

PBruk committed:
Add styling for images with captions and left/right floats.

PBruk committed:
Adjust hero space fade; remove dark border that shows up without image in place.

PBruk committed:
Add border-radius.

PBruk committed:
Add styling for figures and figcaptions.

PBruk committed:
Adjust styling for galleries.

matt-bernhardt committed:
Adds screenshot and README info. Closes #10

matt-bernhardt committed:
Merge pull request #11 from MITLibraries/matt-bernhardt/feature/metadata

Adds screenshot and README info. Closes #10

PBruk committed:
Adds conditional statement to current exhibits loop to display reminder to "...check back often," if no exhibits are current.

PBruk committed:
Remove background that covers hero-image.

PBruk committed:
Change color of link and hover.

PBruk committed:
Make category and category-initial link to about page.

PBruk committed:
Remove code no longer in use.

PBruk committed:
Add button link and styling to exhibits feed.

PBruk committed:
Template was experimental, no longer needed.

PBruk committed:
File deprecated, should inherit from Parent.

PBruk committed:
Add min-width for short child header.

PBruk committed:
Set header_image to be default image for site.

PBruk committed:
Add styling for image attachment page, including previous/next buttons and entry-meta.

PBruk committed:
Add styling for error404 page content.

PBruk committed:
Fix loops to allow all posts from respective query to show.

PBruk committed:
Adjust styling of buttons on hover.

PBruk committed:
Creates new breadcrumb template part to be called on Child front-pages.

PBruk committed:
Standardizes breadcrumbs.

PBruk committed:
Standardizes breadcrumbs; moves while statement to proper place in template.

PBruk committed:
Adjust selector name to fix styling on mobile.

PBruk committed:
Reduce bottom margin between exhibit post content and sidebars on mobile.

PBruk committed:
Fixes columns on mobile and adds margins to sidebars.

PBruk committed:
Adjust weight of title; and line-height on mobile.

PBruk committed:
Update (sidebar) button styles.

PBruk committed:
Add block to header img to fix rendering on FF.

PBruk committed:
Add latest version of Font Awesome via CDN.

PBruk committed:
Header image was being loaded as background-image. We revert to img tag.

PBruk committed:
Remove float from mobile which caused issues on smaller media screens.

PBruk committed:
Adjust font-size and padding on smaller media screens.

PBruk committed:
Adjusts font-size on smaller screens.

PBruk committed:
Adjust styling of Child title on mobile screens.

matt-bernhardt committed:
Merge pull request #12 from MITLibraries/1_travis_testing

Adds Travis and CodeClimate configuration for code analysis

matt-bernhardt committed:
Merge pull request #13 from MITLibraries/basic_styling_pages_and_posts

Merging this branch with the AKDC branch, trying to reduce the amount of uncertainty.

matt-bernhardt committed:
Bumps version to 2.0.0-rc2

matt-bernhardt committed:
Move style.css into css/ directory

Ignore style.css at root (now built by grunt)

Adds 'empty' style.css with warning not to edit

Adds grunt build process for populating version string

matt-bernhardt committed:
Merge pull request #20 from MITLibraries/matt/feature/grunt

Adds grunt build process for version strings

matt-bernhardt committed:
Adds test pattern image (1250x800) for use in testing

frrrances committed:
adjust responsive header and featured header images

matt-bernhardt committed:
Removes hero image to separate template

PBruk committed:
Removes border and excess padding. Resolves #24.

PBruk committed:
Remove max-width. Resolves #24.

PBruk committed:
Undo last commit.

PBruk committed:
Removes max-width.

PBruk committed:
Adds specificity to class.

PBruk committed:
Remove styling. Resolves #30.

PBruk committed:
Change name of class. Resolves #32.

PBruk committed:
Remove styling replaced by use of button class. Resolves #34.

matt-bernhardt committed:
Switches to real default image of Hayden Library

PBruk committed:
Change instructions for widget area.

PBruk committed:
Change dimensions of header-image. Resolves #38.

frrrances committed:
Merge pull request #39 from MITLibraries/phillip/38_change_header_image_dimensions

Change dimensions of header-image.

frrrances committed:
Merge pull request #37 from MITLibraries/phillip/36_change_widget_instructions

Change instructions for the Below Content Widgetized Area

frrrances committed:
Merge pull request #35 from MITLibraries/phillip/34_remove_styling

Remove styling of .disaster-response button

frrrances committed:
Merge pull request #33 from MITLibraries/phillip/32_change_class_name_of_widget

Change class name of widget

frrrances committed:
Merge pull request #29 from MITLibraries/phillip/28_add_specificity_to_class

Adds specificity to class

frrrances committed:
Merge pull request #27 from MITLibraries/phillip/26_make_buttons_full_width_in_sidebars

Removes max-width.

frrrances committed:
Merge pull request #25 from MITLibraries/phillip/24_remove_borders_padding_from_menus_in_widgets

Remove borders and excess padding from widgets that use menus

frrrances committed:
Merge pull request #31 from MITLibraries/phillip/30_remove_styling

Remove .giveNow styling

frrrances committed:
adjust child header styles

frrrances committed:
Merge pull request #44 from MITLibraries/frances/fix/fonts

Adjusts headers size and spacing

matt-bernhardt committed:
Merge pull request #22 from MITLibraries/matt/bug/16_hero_image

Adds default image, implements theme partial for hero image

matt-bernhardt committed:
Automated cleaning of PEAR.Functions.FunctionCallSignature

matt-bernhardt committed:
Ignores .diff files (used for cleaning)

matt-bernhardt committed:
Automated cleaning of Squiz.WhiteSpace.SuperfluousWhitespace

matt-bernhardt committed:
Automated cleaning of WordPress.WhiteSpace.ControlStructureSpacing

matt-bernhardt committed:
Automated cleaning of Generic.WhiteSpace.DisallowSpaceIndent

matt-bernhardt committed:
Automated cleaning of Squiz.Strings.DoubleQuoteUsage

matt-bernhardt committed:
Automated cleaning of Generic.Files.EndFileNewline

matt-bernhardt committed:
Automated cleaning of WordPress.Arrays.ArrayDeclaration

matt-bernhardt committed:
Automated cleaning of Squiz.Commenting.InlineComment

PBruk committed:
Change class names from archnet to alt. Resolves #45.

matt-bernhardt committed:
Automated cleaning of Generic.ControlStructures.InlineControlStructure

matt-bernhardt committed:
Automated cleaning of WordPress.WhiteSpace.OperatorSpacing

matt-bernhardt committed:
Automated cleaning of Generic.Formatting.DisallowMultipleStatements

matt-bernhardt committed:
Automated cleaning of WordPress.WhiteSpace.ControlStructureSpacing

PBruk committed:
Tighten spacing. Resolves #42.

PBruk committed:
Eliminate redundancy.

matt-bernhardt committed:
Automated cleaning of Squiz.Commenting.DocCommentAlignment

matt-bernhardt committed:
Automated cleaning of Generic.Commenting.DocComment

matt-bernhardt committed:
Automated cleaning of Generic.Functions.OpeningFunctionBraceKernighanRitchie

matt-bernhardt committed:
Automated cleaning of Generic.Files.LineEndings

matt-bernhardt committed:
Cleans comments in category-maihaugen-gallery.php

More cleaning of code intenting

Cleans comments on category-rotch-library.php

Fixes comment tags for category.php

Cleans comments in functions.php

Cleans comments in inc/breadcrumb-child.php

Cleans comments in inc/breadcrumbs-sitename.php

Cleans comments in inc/content-front.php

Cleans comments in inc/content-page.php

Cleans comments in inc/content-widgetized.php

Cleans comments in inc/exhibits-current.php

Cleans comments in inc/exhibits-past.php

Cleans comments in inc/exhibits-upcoming.php

Cleans comments (and removes old code) from inc/locationHours.php

Cleans comments in inc/locationInfo.php

matt-bernhardt committed:
Merge pull request #48 from MITLibraries/matt/feature/cleaning

Merging after discussion with @frrrances

matt-bernhardt committed:
Fixes PHP syntax errors

PBruk committed:
Adds widgetized area below content of page template. Resolves #50.

matt-bernhardt committed:
Merge pull request #49 from MITLibraries/matt/bug/php

Fixes PHP syntax errors

frrrances committed:
Merge pull request #46 from MITLibraries/phillip/45_change_class_names

Change class names

frrrances committed:
Merge pull request #47 from MITLibraries/phillip/42_tighten_spacing_akdc_home_page

Tighten spacing.

frrrances committed:
Merge pull request #51 from MITLibraries/phillip/50_widgetize_bottom_front_page_template

Front Page template needs area below content widgetized

frrrances commented on category-maihaugen-gallery.php:
```diff
@@ -0,0 +1,166 @@
+			   		   
+			</div>
+
+		<!---- END OF CURRENT EXHIBITS LOOP --->
```
extra dashes here

frrrances commented on category-maihaugen-gallery.php:
```diff
@@ -0,0 +1,166 @@
+			</div>
+		
+		
+		<!---- END OF UPCOMING EXHIBITS LOOP ----->
```
extra dashes here

frrrances commented on category-maihaugen-gallery.php:
```diff
@@ -0,0 +1,166 @@
+			</div>
+
+
+		<!--- END OF UPCOMING EXHIBITS LOOP --->
```
extra dash

matt-bernhardt committed:
Merge pull request #52 from MITLibraries/matt/feature/comments

Cleans code comments for 14 of the files in the theme. Will do the rest as time allows.

matt-bernhardt commented:
After plenty of work and testing, this branch appears to be ready to merge. Cross your fingers...
